src/main/java/org/dita/dost/reader/GenListModuleReader.java

```diff
@@ -513,10 +520,10 @@ private void handleRootElement(final Attributes atts) {
     public void endElement(final String uri, final String localName, final String qName) throws SAXException {
         // @processing-role
         processRoleStack.pop();
-
-        if (foreignLevel > 0) {
-            foreignLevel--;
-            assert foreignLevel >= 0;
+        
+        final DitaClass cls = classes.pop();
+        if ((cls == null || !cls.isValid()) && !ELEMENT_NAME_DITA.equals(localName)) {
+            return;
```

Comment:
This is a filter. If we return before passing the end element event to the next content handler, we'll end up with unbalanced tree. Is this really correct?